core/src/main/java/com/dsingley/jwt/core/JwtManager.java

```diff
@@ -2,140 +2,90 @@
+ * <p>
+ * Tokens are verified using an {@link Algorithm} provided by the configured {@link VerificationAlgorithmSupplier}
+ * and optionally allowing for clock skew if leeway seconds are configured.
+ */
 @Slf4j
 public class JwtManager {
-    private static final KeyPairGenerator KEY_PAIR_GENERATOR = newKeyPairGenerator();
     private static final JWT JWT_INSTANCE = new JWT();
 
     private final String issuer;
     private final long ttlSeconds;
+    private final SigningAlgorithmSupplier signingAlgorithmSupplier;
+    private final VerificationAlgorithmSupplier verificationAlgorithmSupplier;
     private final long leewaySeconds;
-    private final JwtKeyRepository repository;
-    private final Algorithm algorithm;
-    private final JWTVerifier verifier;
 
     @Builder
     public JwtManager(
             @NonNull String issuer,
             @NonNull Long ttlSeconds,
-            Long leewaySeconds,
-            @NonNull JwtKeyRepository repository
+            @NonNull SigningAlgorithmSupplier signingAlgorithmSupplier,
+            @NonNull VerificationAlgorithmSupplier verificationAlgorithmSupplier,
+            Long leewaySeconds
     ) {
         this.issuer = issuer;
         this.ttlSeconds = ttlSeconds;
+        this.signingAlgorithmSupplier = signingAlgorithmSupplier;
+        this.verificationAlgorithmSupplier = verificationAlgorithmSupplier;
         this.leewaySeconds = leewaySeconds != null && leewaySeconds > 0 ? leewaySeconds : 0;
-        this.repository = repository;
-        Supplier<JwtKeyPair> supplier = Suppliers.memoizeWithExpiration(this::newJwtKeyPair, ttlSeconds, TimeUnit.SECONDS);
-        algorithm = Algorithm.RSA512(new CustomRSAKeyProvider(supplier, this.repository));
-        verifier = JWT.require(algorithm)
-                .withIssuer(issuer)
-                .acceptLeeway(this.leewaySeconds)
-                .build();
     }
 
-    public String create(String subject, Map<String, ?> map) {
+    /**
+     * Create a signed token with the specified subject and claims payload.
+     *
+     * @param subject the subject of the token (required)
+     * @param map the data to include in the token as claims (optional)
+     * @return the signed token
+     */
+    public String create(@NonNull String subject, Map<String, ?> map) {
         String jti = UUID.randomUUID().toString().replace("-", "");
         Instant now = Instant.now();
         String jwt = JWT.create()
                 .withIssuer(issuer)
                 .withJWTId(jti)
                 .withIssuedAt(now)
                 .withNotBefore(now)
-                .withExpiresAt(now.plus(ttlSeconds, ChronoUnit.SECONDS))
+                .withExpiresAt(now.plusSeconds(ttlSeconds))
                 .withSubject(subject)
                 .withPayload(map)
-                .sign(algorithm);
+                .sign(signingAlgorithmSupplier.getAlgorithm());
         log.info("created jti: {} for sub: {}", jti, subject);
```

Comment:
nitpick: we always prefer `LOG` (since the field is `static final`). You do this in a `lombok.config`. Take it or leave it.